components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImplTest.java

```diff
@@ -0,0 +1,360 @@
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        Mockito.when(apiMgtDAO.isEndpointRegistryNameExists(endpointRegistryInfo.getName(), TENANT_ID))
+                .thenReturn(false);
+        Mockito.when(apiMgtDAO.addEndpointRegistry(endpointRegistryInfo, TENANT_ID))
+                .thenReturn(endpointRegistryInfo.getUuid());
+
+        String registryUUID = endpointRegistry.addEndpointRegistry(endpointRegistryInfo);
+
+        Assert.assertEquals(endpointRegistryInfo.getUuid(), registryUUID);
+    }
+
+    @Test(expected = APIMgtResourceAlreadyExistsException.class)
+    public void addEndpointRegistry_existingEntryName() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        Mockito.when(apiMgtDAO.isEndpointRegistryNameExists(endpointRegistryInfo.getName(), TENANT_ID))
+                .thenReturn(true);
+
+        endpointRegistry.addEndpointRegistry(endpointRegistryInfo);
+    }
+
+    @Test
+    public void updateEndpointRegistry() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 1");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        endpointRegistry.updateEndpointRegistry(endpointRegistryInfo.getUuid(), endpointRegistryInfo.getName(),
+                endpointRegistryInfo);
+    }
+
+    @Test(expected = APIMgtResourceAlreadyExistsException.class)
+    public void updateEndpointRegistry_existingEntryName() throws APIManagementException {
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
+        endpointRegistryInfo.setName("Endpoint Registry 2");
+        endpointRegistryInfo.setMode("ReadOnly");
+        endpointRegistryInfo.setType("wso2");
+        endpointRegistryInfo.setOwner(ADMIN_USERNAME);
+        endpointRegistryInfo.setUuid("abc1");
+
+        Mockito.when(apiMgtDAO.isEndpointRegistryNameExists(endpointRegistryInfo.getName(), TENANT_ID))
+                .thenReturn(true);
+
+        endpointRegistry.updateEndpointRegistry(endpointRegistryInfo.getUuid(), "Endpoint Registry 1",
+                endpointRegistryInfo);
+    }
+
+    @Test
+    public void deleteEndpointRegistry() throws APIManagementException {
+        endpointRegistry.deleteEndpointRegistry("abc1");
```

Comment:
let's use Mockito verify

Reply:
updated in https://github.com/wso2/carbon-apimgt/pull/8534/commits/8aad6c2009f4840cedef91797b02feb0a640fa4e